src/vcf/parsing_state.cpp

```diff
@@ -87,5 +91,31 @@ namespace ebi
     {
         defined_metadata.emplace(meta_type, id);
     }
+
+    void ParsingState::validate_additional_checks()
+    {
+        if (additionalChecks.checkEvidence) {
+            if (genotypes_present() || allele_frequencies_present()) {
+                return;
+            }
+            throw new InfoBodyError{record->line, "Genotypes or Allele frequencies are required"};
+        }
+    }
+
+    bool ParsingState::genotypes_present() {
+        auto format = record->format;
+        return !format.empty() && util::contains(format, GT);
+    }
+
+    bool ParsingState::allele_frequencies_present()
+    {
+        auto info = record->info;
+        for (auto &in : info) {
+            if (in.first == vcf::AF) {
```

Comment:
`auto info = record->info` is making a copy. Either use a reference `auto &info = ...` or just use it inlined. In this case I think the variable doesn't add clarity.
```suggestion
        for (const auto &info_key_value : record->info) {
            if (info_key_value.first == vcf::AF) {
```
also, put `const` in every read-only reference.